Jenkinsfile / test / tests: Ginkgo validation

ianvernon opened the pull request:
This branch contains fixes that enable that the Ginkgo build framework functions appropriately on the Cilium Jenkins setup. Some other minor enhancements / test gaps were fixed, specifically running `go vet`, and outputting `stderr` of Vagrant commands.

The Jenkinsfile for the Ginkgo tests has been edited to run only on build slaves with the label 'ginkgo'; this is intentional.

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
Jenkinsfile: update Ginkgo Jenkinsfile

* Update agent for Ginkgo Jenkinsfile to run on test Jenkins slave
* Output environment variables

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
Makefile: run go vet in tests-common-ginkgo

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
tests: do not fail 00-fmt.sh when finding files

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test: fix typo in Vagrant error message

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test/helpers: get stderr of Vagrant commands

Signed-off by: Ian Vernon <ian@cilium.io>

tgraf commented on tests/00-fmt.sh:
```diff
@@ -2,12 +2,14 @@
 
 set -ex
 
+set +e
```
The `set -e` and `set +e` in a row is weird

aanm commented on tests/00-fmt.sh:
```diff
@@ -1,13 +1,14 @@
 #!/usr/bin/env sh
 
-set -ex
+set -x
```
What's failing on the diff that make you move the `set -e` to line 11?

ianvernon replied:
the bindata file does not exist when this is ran in the Ginkgo suite. When we were running this in the current build setup, it does exist. Since we are inspecting the output of the command, we don't need `set -e` for it anyway.

Commit:
Vagrantfile: ignore src directory with rsync

When running the bash script runtime tests VM, the VM for the runtime tests
tries to rsync the entire Cilium directory. The new Ginkgo tests create a
symlink within the Cilium directory that links back to the root of the Cilium
directory. This creates a circular link. When the bash script runtime tests VM
boots, it gets in a circular loop when trying to mount the code at the symlink,
and runs out of space. Ignore this directory when booting the bash script
runtime tests VM.

Signed-off by: Ian Vernon <ian@cilium.io>

joestringer commented on test/helpers/vagrant.go:
```diff
@@ -53,12 +53,24 @@ func (vagrant *Vagrant) Create(scope string) error {
-			log.Infof(in.Text()) // write each line to your log
+			log.Infof("stdout: %s", in.Text()) // write each line to your log
+		}
+	}()
```
Can we just use io.Copy() like the ssh helpers do?

ianvernon replied:
Good idea! Will do.

ianvernon replied:
Discussed offline; we will leave this as is in the review for now.

Commit:
tests: ignore 00-fmt in check flags

Signed-off by: Ian Vernon <ian@cilium.io>
